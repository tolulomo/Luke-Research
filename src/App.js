import React, { Component } from 'react';
import { BrowserRouter, Route } from 'react-router-dom';

const Header = () => <div>Header</div>
const Welcome = () => <div>Welcome</div>
const Project = () => <div>Project</div>
const AddProject = () => <div>Add Project</div>
const AddUser = () => <div>Add User Page</div>


class App extends Component {
  render() {
    return (
      <BrowserRouter>
        <div>
          <Header />
          <Route exact path="/" component={Welcome} />
          <Route exact path="/project" component={Project} />
          <Route path="/project/new" component={AddProject} />
          <Route path="/project/add_user" component={AddUser} />
        </div>
      </BrowserRouter>
    );
  }
}

export default App;

// import React from "react";
// import "./App.css";

// function App() {
//   return (
//     <div className="App">
//       <header className="App-header">
//         <h3>
//           Let's create a view that displays user data grouped by the project
//           each user is working on.
//         </h3>
//         <p>
//           Start by looking at <code>src/mock_data/users.js</code> and{" "}
//           <code>src/mock_data/projects.js</code>.
//         </p>
//       </header>
//     </div>
//   );
// }

// export default App;
