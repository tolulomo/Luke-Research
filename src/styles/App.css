@import 'abstract/_animation.css';
@import 'abstract/_mixin.css';
@import 'abstract/_variable.css';
@import 'base/_global.css';
@import 'base/_grid.css';
@import 'base/_utility.css';
@import 'components/_floatingBtn.css';
@import 'components/_header.css';
@import 'components/_welcomePage.css';