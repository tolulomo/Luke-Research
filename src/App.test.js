import React from "react";
import ReactDOM from 'react-dom';
import { render } from "@testing-library/react";
import renderer from 'react-test-renderer';
import App from "./App";

// test("renders explanation text and starting directions", () => {
//   const { getByText } = render(<App />);
//   const explanationTextHeader = getByText(
//     /Let's create a view that displays user data grouped by the project each user is working on/i
//   );
//   const directionsElement = getByText(/Start by looking at /i);
//   expect(explanationTextHeader).toBeInTheDocument();
//   expect(directionsElement).toBeInTheDocument();
// });

it('renders wihout crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('App', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<App />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

