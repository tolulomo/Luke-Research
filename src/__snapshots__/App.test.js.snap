// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot renders 1`] = `
<div>
  <div>
    Header
  </div>
  <div>
    Welcome
  </div>
</div>
`;
